examples/applications/plot_time_series_lagged_features.py

```diff
@@ -0,0 +1,433 @@
+print("========================")
+
+# %%
+# Next, we take a look at the statistical summary of the dataset
+# so that we can better understand the data that we are working with.
+summary = pd.DataFrame(df.describe())
+summary = (
+    summary.style.background_gradient()
+    .set_table_attributes("style = 'display: inline'")
+    .set_caption("Statistics of the Dataset")
+    .set_table_styles([{"selector": "caption", "props": [("font-size", "16px")]}])
+)
+summary
+
+# %%
+# Taking a look at the total count of rented bikes for the first
+# week in our dataset.
+import matplotlib.pyplot as plt
+import seaborn as sns
+
+bike_count = df["count"]
+bike_count[: 7 * 24].plot(figsize=(15, 6))
+_ = plt.title("First week of bike rental count data")
+
+# %%
+# Next, let us look at the fraction of the rented fleet vs the number
+# number of hours.
+fig, ax = plt.subplots(figsize=(12, 4))
+bike_count.hist(bins=30, ax=ax)
+_ = ax.set(
+    xlabel="Fraction of rented fleet demand",
+    ylabel="Number of hours",
+)
+
+# %%
+# Let us look at the count of the seasons `fall`, `spring`, `summer`
+# and `winter` present in the dataset.
+plt.figure(figsize=(15, 10))
+ax = sns.countplot(data=df, x="season")
+
+bbox_args = dict(boxstyle="round", fc="0.9")
+for p in ax.patches:
+    ax.annotate(
+        "{:.0f}".format(p.get_height()),
+        (p.get_x() + 0.3, p.get_height() + 75),
+        color="black",
+        bbox=bbox_args,
+        fontsize=15,
+    )
+
+plt.title("Count for Season", fontsize=20)
+plt.xlabel("Season", fontsize=15)
+plt.ylabel("Count", fontsize=15)
+plt.xticks(fontsize=10)
+plt.yticks(fontsize=10)
+plt.show()
+
+# %%
+# To check for outliers in the features of the dataset we create box-plots
+# for `temp`, `feel_temp`, `humidity` and `windspeed`.
+plt.figure(figsize=(15, 15))
+
+
+def create_boxplot(feature, color):
+    sns.boxplot(data=df[[feature]], color=color)
+    plt.title("Box Plot for " + feature.title(), fontsize=20)
+    plt.xticks(fontsize=10)
+    plt.yticks(fontsize=10)
+
+
+plt.subplot(221)
+create_boxplot("temp", "blue")
+
+plt.subplot(222)
+create_boxplot("feel_temp", "orange")
+
+plt.subplot(223)
+create_boxplot("humidity", "green")
+
+plt.subplot(224)
+create_boxplot("windspeed", "red")
+
+# %%
+# Exploring the hourly count of bikes over weekdays:
+plt.figure(figsize=(15, 10))
+sns.pointplot(x=df["hour"].astype(int), y=df["count"], hue=df["weekday"].astype(int))
+plt.title("Hourly Count for Weekdays", fontsize=20)
+plt.xlabel("Hour", fontsize=15)
+plt.ylabel("Count", fontsize=15)
+plt.xticks(fontsize=10)
+plt.yticks(fontsize=10)
+plt.show()
+
+# %%
+# Exploring the hourly count of bikes over different seasons:
+plt.figure(figsize=(15, 10))
+sns.pointplot(x=df["hour"].astype(int), y=df["count"], hue=df["season"])
+plt.title("Hourly Count for Seasons", fontsize=20)
+plt.xlabel("Hour", fontsize=15)
+plt.ylabel("Count", fontsize=15)
+plt.xticks(fontsize=10)
+plt.yticks(fontsize=10)
+plt.show()

```

Comment:
This two plots are quite interesting to explore the data! Following the logic of referencing the [Time-related feature engineering example](https://scikit-learn.org/stable/auto_examples/applications/plot_cyclical_feature_engineering.html) for data exploration, I suggest you open a PR to rather add this plots on that notebook.

Just keep in mind that "weekdays" is not the same as "days of the week" :)